hex/Game.php

```
@@ -0,0 +1,118 @@
+<?php
+
+namespace Hex;
+
+
+/**
+ * Class Game
+ * @package Hex
+ */
+class Game
+{
+
+    public const BOARD_MINI_SIZE = 5;
+    public const BOARD_DEFAULT_SIZE = 11;
+    public const BOARD_LARGE_SIZE = 15;
+
+    protected $id;
+
+    protected $stones;
+
+    protected $size;
+
+    /**
+     * Game constructor.
+     * @param $size
+     */
+    public function __construct($size)
+    {
+        $this->size = $size;
+        $this->stones = [];
+    }
+
+    /**
+     * @param $x
+     * @param $y
+     * @return bool
+     */
+    public function hasStone($x, $y)
+    {
+        return !empty(array_filter($this->stones, function ($stone) use ($x, $y) {
+            return $x === $stone[0] && $y === $stone[1];
+        }));
+    }
+
+    /**
+     * @param $x
+     * @param $y
+     */
+    public function addStone(int $x, int $y)
+    {
+        $limit_min = 0;
+        $limit_max = $this->getSize() - 1;
+
+        $x_is_inside = $x >= $limit_min && $x <= $limit_max;
+        $y_is_inside = $y >= $limit_min && $y <= $limit_max;
+        $already_played_move = in_array([$x, $y], $this->stones);
+
+        if ($x_is_inside && $y_is_inside && !$already_played_move) {
```

If the stone is out of bould, you should throw an Exception. Please migrate this logic in a specific class in next PR's